src/native/libs/System.Globalization.Native/pal_collation.m

```diff
@@ -269,4 +270,83 @@ int32_t GlobalizationNative_EndsWithNative(const uint16_t* localeName, int32_t l
     return result == NSOrderedSame ? 1 : 0;
 }
 
+/**
+ * Append a code point to a string, overwriting 1 or 2 code units.
+ * The offset points to the current end of the string contents
+ * and is advanced (post-increment).
+ * "Safe" macro, checks for a valid code point.
+ * If a surrogate pair is written, checks for sufficient space in the string.
+ * If the code point is not valid or a trail surrogate does not fit,
+ * then isError is set to true.
+ *
+ * @param s const UChar * string buffer
+ * @param i string offset, must be i<capacity
+ * @param capacity size of the string buffer
+ * @param c code point to append
+ * @param isError output UBool set to true if an error occurs, otherwise not modified
+ * @stable ICU 2.4
+ */
+#define Append(s, i, capacity, c, isError) UPRV_BLOCK_MACRO_BEGIN { \
+    if((uint32_t)(c)<=0xffff) { \
+        (s)[(i)++]=(uint16_t)(c); \
+    } else if((uint32_t)(c)<=0x10ffff && (i)+1<(capacity)) { \
+        (s)[(i)++]=(uint16_t)(((c)>>10)+0xd7c0); \
+        (s)[(i)++]=(uint16_t)(((c)&0x3ff)|0xdc00); \
+    } else /* c>0x10ffff or not enough space */ { \
+        (isError)=true; \
+    } \
+} UPRV_BLOCK_MACRO_END
+
+/*
+Function:
+ChangeCaseNative
+
+Returns upper or lower casing of a string, taking into account the specified locale.
+*/
+int32_t GlobalizationNative_ChangeCaseNative(const uint16_t* localeName, int32_t lNameLength,
```

Comment:
Maybe it is also possible to re-use: https://github.com/dotnet/runtime/blob/2e764d6eb092fc8f05c42e6f137347c3eab45fd0/src/coreclr/pal/src/locale/unicodedata.cpp#L11 here by moving it to `src/native/minipal`?
cc @jkotas, @lambdageek

Reply:
It would go against the spirit of this feature. This feature is trying to make the globalization implementation minimal by relying on the OS provided globalization.

Reply:
Ah, I thought it is converting the string via the macro added few lone above. :)